# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [10]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [11]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.9726726 0.5288531 0.3179636 0.9972922 0.7351041 0.3423312 0.6290019
2 0.1966151 0.8359006 0.2136413 0.2212330 0.1580872 0.6204461 0.1766443
  V8        V9        V10       V11        V12       V13       V14      
1 0.4701345 0.3039726 0.8468796 0.35216377 0.7486217 0.6739353 0.3675967
2 0.2787185 0.3230176 0.6198060 0.02558903 0.3899260 0.1194005 0.7518124
  V15       V16       V17       V18       V19       V20       
1 0.3941176 0.1914709 0.7671313 0.3786483 0.9204706 0.09378575
2 0.7408187 0.5965192 0.4198766 0.4094285 0.6252443 0.12991165

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [12]:
df2 <- df1 %>% gather(key='Key',value='Value')
  

head(df2)

Key Value    
1 V1  0.9726726
2 V1  0.1966151
3 V2  0.5288531
4 V2  0.8359006
5 V3  0.3179636
6 V3  0.2136413

## Assignment 2

We will generate some data:

In [13]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [14]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [15]:
df5 <- full_join(df3,df4) 
head(df5)

Joining, by = "id"


id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook